corehq/apps/userreports/reports/filters/specs.py

```diff
@@ -76,6 +76,21 @@ class FilterSpec(JsonObject):
     def get_display(self):
         return self.display or self.slug
 
+    @classmethod
+    def wrap(cls, *args, **kwargs):
+        # Because pre_value is set to DefaultProperty, pre_value is automatically
+        # interpreted to be a datatype when it's fetched by jsonobject.
+        # This fails hard when the spec expects a string
+        # and gets something else...like a date.
+        doc = args[0]
+        if 'pre_value' in doc:
+            pre_value = doc['pre_value']
+            data_type = doc['datatype']
+            if data_type == 'string' and type(pre_value) != 'str':
```

Comment:
Shouldn't the second check not have quotes around `str`? Or you could drop the second half of the conditional altogether, since there's no harm in re-casting a string to a string.

Reply:
lol thank u. I think when i wrote this line my brain was thinking of doing something else after but then switched to just casting the value to a string